freqtrade/optimize/backtesting.py

```diff
@@ -218,21 +218,18 @@ def _get_ohlcv_as_lists(self, processed: Dict[str, DataFrame]) -> Dict[str, Tupl
         """
         # Every change to this headers list must evaluate further usages of the resulting tuple
         # and eventually change the constants for indexes at the top
-        headers = ['date', 'buy', 'open', 'close', 'sell', 'low', 'high']
+        headers = ['date', 'buy', 'open', 'close', 'sell', 'low', 'high', 'buy_tag']
         data: Dict = {}
         self.progress.init_step(BacktestState.CONVERT, len(processed))
 
         # Create dict with data
         for pair, pair_data in processed.items():
             self.check_abort()
             self.progress.increment()
-            has_buy_tag = 'buy_tag' in pair_data
-            headers = headers + ['buy_tag'] if has_buy_tag else headers
```

Comment:
I guess the correct approach here will be to move this part down below the `advise_*` calls.

Removing it will however break backwards compatibility (force all strategies to use buy_tag).